pkg/cmd/pulumi/util_remote.go

```diff
@@ -171,6 +172,65 @@ type RemoteArgs struct {
 	executorImagePassword    string
 }
 
+func (r *RemoteArgs) applyFlagsForDeploymentCommand(cmd *cobra.Command) {
+	cmd.PersistentFlags().BoolVar(
+		&r.inheritSettings, "inherit-settings", true,
+		"Inherit deployment settings from the current stack")
+	cmd.PersistentFlags().BoolVar(
+		&r.streamLogs, "stream-logs", false,
+		"Suppress log streaming of the deployment job")
+	cmd.PersistentFlags().StringArrayVar(
+		&r.envVars, "env", []string{},
+		"Environment variables to use in the remote operation of the form NAME=value "+
+			"(e.g. `--env FOO=bar`)")
+	cmd.PersistentFlags().StringArrayVar(
+		&r.secretEnvVars, "env-secret", []string{},
+		"Environment variables with secret values to use in the remote operation of the form "+
+			"NAME=secretvalue (e.g. `--env FOO=secret`)")
+	cmd.PersistentFlags().StringArrayVar(
+		&r.preRunCommands, "pre-run-command", []string{},
+		"Commands to run before the remote operation")
+	cmd.PersistentFlags().BoolVar(
+		&r.skipInstallDependencies, "skip-install-dependencies", false,
+		"Whether to skip the default dependency installation step")
+	cmd.PersistentFlags().StringVar(
+		&r.gitBranch, "git-branch", "",
+		"Git branch to deploy; this is mutually exclusive with --git-commit; "+
+			"either value needs to be specified")
+	cmd.PersistentFlags().StringVar(
+		&r.gitCommit, "git-commit", "",
+		"Git commit hash of the commit to deploy (if used, HEAD will be in detached mode); "+
```

Comment:
no, those are parameters sent when the deployment job is created. This is currently supported by the automation api and now we are exposing it in a command

https://github.com/pulumi/pulumi/blob/f69f1d2ee3cdc98c56aaad0094c5b474a1a062df/pkg/cmd/pulumi/util_remote.go#L204-L207

Reply:
```suggestion
		"Git commit hash of the commit to deploy; "+
```

Great, I don't think we need to specify `HEAD will be in detached mode` then, since this should not be relevant to the user.  They just need to know that they'll deploy this particular commit hash.  In fact, I think we should also remove this text in `util_remote.go`, though that would be in a separate PR.

Reply:
well `HEAD will be in detached mode` is still true for the deployment, it will affect the program running there (the program running in the deployment job)

Reply:
fair enough!